src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13957,6 +14855,169 @@ public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y)
             public static Vector512<T> Invoke(Vector512<T> x) => (x * T.CreateChecked(180)) / T.Pi;
         }
 
+        /// <summary>T &lt;&lt; amount</summary>
+        internal readonly struct ShiftLeftOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IShiftOperators<T, int, T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable => true;
+
+            public T Invoke(T x) => x << _amount;
+            public Vector128<T> Invoke(Vector128<T> x) => x << _amount;
+            public Vector256<T> Invoke(Vector256<T> x) => x << _amount;
+            public Vector512<T> Invoke(Vector512<T> x) => x << _amount;
+        }
+
+        /// <summary>T &gt;&gt; amount</summary>
+        internal readonly struct ShiftRightArithmeticOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IShiftOperators<T, int, T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable => true;
+
+            public T Invoke(T x) => x >> _amount;
+            public Vector128<T> Invoke(Vector128<T> x) => x >> _amount;
+            public Vector256<T> Invoke(Vector256<T> x) => x >> _amount;
+            public Vector512<T> Invoke(Vector512<T> x) => x >> _amount;
+        }
+
+        /// <summary>T &gt;&gt;&gt; amount</summary>
+        internal readonly struct ShiftRightLogicalOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IShiftOperators<T, int, T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable => true;
+
+            public T Invoke(T x) => x >>> _amount;
+            public Vector128<T> Invoke(Vector128<T> x) => x >>> _amount;
+            public Vector256<T> Invoke(Vector256<T> x) => x >>> _amount;
+            public Vector512<T> Invoke(Vector512<T> x) => x >>> _amount;
+        }
+
+        /// <summary>T.RotateLeft(amount)</summary>
+        internal readonly struct RotateLeftOperator<T>(int amount) : IStatefulUnaryOperator<T> where T : IBinaryInteger<T>
+        {
+            private readonly int _amount = amount;
+
+            public static bool Vectorizable => true;
+
+            public T Invoke(T x) => T.RotateLeft(x, _amount);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (sizeof(T) == 1) return (Vector128.ShiftLeft(x.AsByte(), _amount & 7) | (Vector128.ShiftRightLogical(x.AsByte(), (8 - _amount) & 7))).As<byte, T>();
```

Comment:
The masking should be unnecessary, `Generic Math` and `VectorXXX<T>` types already mask to `(sizeof(T) * 8) - 1` bits to ensure consistent and deterministic results.

We can just do `(x << _amount) | (x >>> ((sizeof(T) * 8) - _amount)`.

For x64, we can optionally optimize to use `Avx512F.RotateLeft` (V512) and `Avx512F.VL.RotateLeft` (V128/V256) or we can just wait for the `Vector128.RotateLeft` APIs to be exposed and defer to that for .NET 9

Reply:
(I'll just leave that out for now)

Reply:
> Should I suppress those?

Ah, I see.

We exposed `, byte count` but didn't expose `, Vector128<T> count`. Internally we do the right thing and transform non constant `count` into the latter; but really we should've exposed both for consistency with `Sse2.Shift*` and to ensure users didn't unnecessarily get the warning.

We should just expose the set for parity.

> (I'll just leave that out for now)

Sounds fine, we can update in a follow up as part of adding the `Vector128.RotateLeft/Right` APIs.